lib/routes/highwaterApi.js

```diff
@@ -47,22 +48,43 @@ module.exports = function (userApiClient, kissClient, salt) {
      IMPLEMENTATIONS OF METHODS
      */
     named_user: function (req, res, next) {
+      message.user = hash_id(req.params.userid, 10)
+      message.eventName = req.params.eventname
+      message.event = req.query
+      message.params = req.params
+      message.url = req.url
+      message.method = req.method
+      console.log(message.user + " in " + message.eventName)
       log.debug('named_user: params[%j], url[%s], method[%s]', req.params, req.url, req.method);
-      kissClient.submit(hash_id(req.params.userid, 10), req.params.eventname, req.query);
+      eventsProducer.createEventsHandler(message)
```

Comment:
I believe you should instantiate the events handler one level above and pass it as an argument. Otherwise, the code here will instantiate a new events handler for every request.